### PR TITLE
Godepchecker

### PR DESCRIPTION
**Pull request by jim-minter**
Added a checker to ensure s2i's godeps match origin's.  Currently, warnings are printed but do not cause make verify to bomb out.
Bumped two vendorings to match origin as well.

@soltysh

### CODE REVIEW

**Comment by soltysh** [comment]
I was rather thinking about verifying the godeps if the are valid, and not if they are the same as in origin. Iow. if `godep restore` works cleanly without any errors.

File: hack/common.sh [hunk 1]
```diff
@@ -293,7 +293,9 @@ s2i::build::place_bins() {
 
       # Move the specified release binaries to the shared S2I_OUTPUT_BINPATH.
       for binary in "${binaries[@]}"; do
-        mv "${full_binpath_src}/${binary}" "${S2I_OUTPUT_BINPATH}/${platform}/"
+        if [ -e "${full_binpath_src}/${binary}" ]; then
+          mv "${full_binpath_src}/${binary}" "${S2I_OUTPUT_BINPATH}/${platform}/"
+        fi
```
Comment (bparees) [inline hack/common.sh]: don't we want to error if a binary we expected to build doesn't exist?

File: tools/godepchecker/godepchecker.go [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+}
+
+func readJSON(filename string) (godeps *Godeps, err error) {
+	f, err := os.Open(filename)
+	if err != nil {
+		return
+	}
+	defer f.Close()
+
+	json.NewDecoder(f).Decode(&godeps)
```
Comment (jim-minter) [inline tools/godepchecker/godepchecker.go]: Yes.

**Comment by bparees** [comment]
lgtm
[merge]

**Pull request merged**